### PR TITLE
backend/feat: #3403 clean up of driver location in dynamic offer bpp

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs [hunk 1]
```diff
@@ -364,7 +354,6 @@ cancelBooking booking mbDriver transporter = do
   QBCR.upsert bookingCancellationReason
   whenJust mbRide $ \ride -> do
     QRide.updateStatus ride.id DRide.CANCELLED
-    DLoc.updateOnRide booking.providerId ride.driverId False
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs:367]: @chakri642 We are removing OnRide update in Location Table. Then where are we calling rideDetails with status CANCELLED ? Somewhere in a common function, can you verify if this flow would also reach to that function and it is handled already ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Booking.hs [hunk 2]
```diff
@@ -70,13 +66,6 @@ stuckBookingsCancel merchantShortId req = do
   let stuckPersonIds = stuckRideItems <&> (.driverId)
   let stuckDriverIds = cast @DP.Person @DP.Driver <$> stuckPersonIds
   -- drivers going out of ride, update location from redis to db
-  enableLocationTrackingService <- asks (.enableLocationTrackingService)
-  driverLocations <-
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Booking.hs:74]: @chakri642 We have removed this code considering there is no more Redis Going to be there for location updates right so stuck booking does not need to do anything with location updates right ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 3]
```diff
@@ -911,14 +910,13 @@ getPaymentHistoryEntityDetails merchantShortId driverId invoiceId = do
...
 clearOnRideStuckDrivers merchantShortId dbSyncTime = do
-  merchant <- findMerchantByShortId merchantShortId
+  _merchant <- findMerchantByShortId merchantShortId
   now <- getCurrentTime
   let dbSyncInterVal = addUTCTime (fromIntegral (- fromMaybe 1 dbSyncTime) * 60) now
   driverInfos <- B.runInReplica $ QPerson.getOnRideStuckDriverIds dbSyncInterVal
   driverIds <-
     mapM
       ( \driverInf -> do
-          void $ DLoc.updateOnRide merchant.id driverInf.driverId False
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:921]: @chakri642 Here we should handle by calling our rideDetails API with status CANCELLED

Comment (khuzema786) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:921]: Unless if it is already getting handled by the Ride Cancel flow where you have called LTS rideDetails API

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs [hunk 4]
```diff
@@ -93,21 +93,21 @@ data DriverLocationUpdateStreamData = DriverLocationUpdateStreamData
   }
   deriving (Generic, FromJSON, ToJSON)
 
-buildUpdateLocationHandle ::
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:96]: Lets Delete This

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide/Internal.hs [hunk 5]
```diff
@@ -146,20 +140,16 @@ cancelRideImpl rideId bookingCReason = do
...
   Id SRB.Booking ->
   DRide.Ride ->
   SBCR.BookingCancellationReason ->
   Id DMerc.Merchant ->
   m ()
-cancelRideTransaction bookingId ride bookingCReason merchantId = do
+cancelRideTransaction bookingId ride bookingCReason _merchantId = do
   let driverId = cast ride.driverId
   driverInfo <- QDI.findById (cast ride.driverId) >>= fromMaybeM (PersonNotFound ride.driverId.getId)
-  void $ DLoc.updateOnRide merchantId ride.driverId False
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide/Internal.hs:162]: @chakri642 Here Also we should call our Ride Cancelled API if not already handled by our common CancelRideImpl function.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs [hunk 6]
```diff
@@ -117,7 +112,6 @@ endRideTransaction ::
   Id Merchant ->
   m ()
 endRideTransaction driverId booking ride mbFareParams mbRiderDetailsId newFareParams thresholdConfig merchantId = do
-  DLoc.updateOnRide merchantId ride.driverId False
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs:120]: @chakri642 Here also can we check if Ride Cancelled of LTS service is called somewhere in the later flow or previous flow.

Comment (chakri642) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs:120]: Handled.
```tripEndPoints <- do
            res <- LF.rideEnd rideId tripEndPoint.lat tripEndPoint.lon booking.providerId driverId
            pure $ toList res.loc```

**Review by khuzema786 — APPROVED** [review]

**Review by khuzema786 — DISMISSED** [review]

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**